actors/miner/src/lib.rs

```diff
@@ -4652,6 +4869,119 @@ fn validate_precommits(
     Ok((batch.gen(), verify_infos))
 }
 
+fn validate_ni_sectors(
+    rt: &impl Runtime,
+    sectors: &[SectorNIActivationInfo],
+    seal_proof_type: RegisteredSealProof,
+    all_or_nothing: bool,
+) -> Result<(BatchReturn, Vec<SectorSealProofInput>), ActorError> {
+    let receiver = rt.message().receiver();
+    let miner_id = receiver.id().unwrap();
+    let curr_epoch = rt.curr_epoch();
+    let activation_epoch = curr_epoch;
+    let challenge_earliest = curr_epoch - rt.policy().max_prove_commit_ni_randomness_lookback;
+    let unsealed_cid = CompactCommD::empty().get_cid(seal_proof_type).unwrap();
+    let entropy = serialize(&receiver, "address for get verify info")?;
+
+    if sectors.is_empty() {
+        return Ok((BatchReturn::empty(), vec![]));
+    }
+    let mut batch = BatchReturnGen::new(sectors.len());
+
+    let mut verify_infos = vec![];
+    let mut sector_numbers = BitField::new();
+    for (i, sector) in sectors.iter().enumerate() {
+        let mut fail_validation = false;
+
+        let set = sector_numbers.get(sector.sector_number);
+        if set {
+            warn!("duplicate sector number {}", sector.sector_number);
+            fail_validation = true;
```

Comment:
I think it would be appropriate to just abort in this case (illegal argument). If a sector number is duplicated, there is some ambiguity about which one should be activated. They're not quite fungible (different expiration epochs). PreCommitSector also aborts with duplicates.

Reply:
Question posed in slack (prompted by the `seal_rand_epoch >= curr_epoch` case below) about which other cases we should abort on: https://filecoinproject.slack.com/archives/C06EZQ2DTD0/p1718871885587219